# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/GPUVideoImage.h
# Commit: 2a830d3c1fb0
# Full Hash: 2a830d3c1fb01d887bd9ec41aeb8b51dbc0bfc62
# Author: Brad Werth <bwerth@mozilla.com>
# Date: 2022-09-09 04:30:58
# Regressor Bug: 1787561
# File Overlap Count: 1
# Description:
#   Bug 1787561 Part 1: Make GPUVideoImage track color depth, and make RemoteImageHolder pass it through. r=media-playback-reviewers,gfx-reviewers,alwu,aosmond
#   
#   This ensures that the Image that the client receives has the color depth
#   information that was sent to the GPU, even though the client can't query
#   the GPU directly.
# ==============================================================================

diff -r edaed3aeb0f9 -r 2a830d3c1fb0 gfx/layers/GPUVideoImage.h
--- a/gfx/layers/GPUVideoImage.h	Thu Sep 08 17:55:07 2022 +0000
+++ b/gfx/layers/GPUVideoImage.h	Thu Sep 08 18:02:03 2022 +0000
@@ -38,8 +38,11 @@
 
  public:
   GPUVideoImage(IGPUVideoSurfaceManager* aManager,
-                const SurfaceDescriptorGPUVideo& aSD, const gfx::IntSize& aSize)
-      : Image(nullptr, ImageFormat::GPU_VIDEO), mSize(aSize) {
+                const SurfaceDescriptorGPUVideo& aSD, const gfx::IntSize& aSize,
+                const gfx::ColorDepth& aColorDepth)
+      : Image(nullptr, ImageFormat::GPU_VIDEO),
+        mSize(aSize),
+        mColorDepth(aColorDepth) {
     // Create the TextureClient immediately since the GPUVideoTextureData
     // is responsible for deallocating the SurfaceDescriptor.
     //
@@ -56,6 +59,8 @@
 
   gfx::IntSize GetSize() const override { return mSize; }
 
+  gfx::ColorDepth GetColorDepth() const override { return mColorDepth; }
+
   Maybe<SurfaceDescriptor> GetDesc() override {
     return GetDescFromTexClient(mTextureClient);
   }
@@ -89,6 +94,7 @@
 
  private:
   gfx::IntSize mSize;
+  gfx::ColorDepth mColorDepth;
   RefPtr<TextureClient> mTextureClient;
 };
 
